src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -264,16 +263,14 @@ protected MachineTranslationInfo search() throws Exception {
 
         private String getTranslation(Language source, Language target) {
             if (!force) {
-                if (!Preferences.isPreferenceDefault(Preferences.MT_AUTO_FETCH, true)) {
-                    return translator.getCachedTranslation(source, target, src);
-                }
-                if (Preferences.isPreference(Preferences.MT_ONLY_UNTRANSLATED)) {
-                    TMXEntry entry = Core.getProject().getTranslationInfo(currentlyProcessedEntry);
-                    if (entry.isTranslated()) {
-                        return translator.getCachedTranslation(source, target, src);
-                    }
+                String cached = translator.getCachedTranslation(source, target, src);
+                if (cached != null || !Preferences.isPreferenceDefault(Preferences.MT_AUTO_FETCH, false)
```

Comment:
We can check code coverage whether the all the conditions are covered in unit tests.